### DISCUSSION TITLE
volt - text_area & htmlspecialchars

### DISCUSSION THREAD

**Discussion by davidcorreia** [phalcon/cphalcon#15991] [General]
I have been testing phalcon v5 (RC1 and just now RC2) with php8.1 and i keep getting warnings like

`PHP Deprecated:  htmlspecialchars(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/html/cache/_var_www_html_app_views_index.volt.php on line 453`

when i edit the "compiled" file what i see in line 453 is this:
` <?= \Phalcon\Tag::textArea(['suggest', 'class' => 'form-control', 'style' => 'height:100px']) ?>`
origin code is `{{ text_area("suggest", 'class':'form-control', 'style':'height:100px') }}`

after i replaced it to simple html the warnings disappeared.
Also if, `Tag::setDefault('suggest', '');` is set in the controller, the warning also disappeared.

its a simple workaround, but is it the expected behavior?

**Comment by niden** [comment]
`xxxx)_: Passing null to parameter #1....` will be appearing more in the near future for sure. This change for PHP 8.x will definitely show areas of the framework that we need to ensure that whatever is passed is a string and not `null`. In the past PHP was handling all that, casting a `null` to an empty string. From what I read, this will be removed in PHP 9.

You could for now use the `TagFactory` to create your code. The `setDefaults()` in `Tag` was always buggy, this is why we rewrote `Tag`.

If you don't mind, can you please open an issue so that I can add a workaround ?

**Reply by davidcorreia** [reply]
thank you, just did.
